pkg/runtime/runtime.go

```diff
@@ -386,8 +386,9 @@ func (a *DaprRuntime) initRuntime(opts *runtimeOpts) error {
 			log.Warnf("failed to watch component updates: %s", err)
 		}
 	}
-
-	a.appendBuiltinSecretStore()
+	if a.runtimeConfig.EnableBuiltinSecretStore {
+		a.appendBuiltinSecretStore()
```

Comment:
Please, create a new E2E test to validate that the k8s secret store can be enabled/disabled correctly.

Reply:
Generally I wouldn't require an E2E test for something so small in scope, but in this case there's also the Injector component involved, and we need to ensure backwards compatibility all the way from injection, so E2E makes sense here.

Reply:
Well ACTUALLY... backwards compatibility is tested by the virtue of having the existing e2e pass.

Reply:
That is true. But we should have an E2E test showing that this feature works. The E2E test should have a scenario targeting reading from K8s secrets directly. One for on and another one for off.

Reply:
We already have `secretapp` taking care of default case where-in built-in secret store named `kubernetes` is used. I have added further case in the same test to take care of disabled k8s secret store.